drivers/clock_control/nrf_power_clock.c

```diff
@@ -424,11 +420,11 @@ void nrf_power_clock_isr(void *arg)
 
 	if (clock_event_check_and_clean(NRF_CLOCK_EVENT_LFCLKSTARTED,
 					NRF_CLOCK_INT_LF_STARTED_MASK)) {
+		clkstarted_handle(dev, CLOCK_CONTROL_NRF_TYPE_LFCLK);
 		if (IS_ENABLED(
 			CONFIG_CLOCK_CONTROL_NRF_K32SRC_RC_CALIBRATION)) {
 			z_nrf_clock_calibration_lfclk_started();
 		}
-		clkstarted_handle(dev, CLOCK_CONTROL_NRF_TYPE_LFCLK);
```

Comment:
I don't entirely understand the motivation of this change. I thought the calibration was to be started ASAP after the LF clock gets running, before any user callbacks. Why was the order inverted?

Reply:
the problem is that calibration is using system work queue which is initialized in post kernel so when LF clock is started before system work queue is ready then error occurs. Configuration provided in the issue was provoking this case.

I don't like this solution and it would be better to not rely on work queue but for that #21667 would need to be merged and that won't happen for that release.

Reply:
Yes, I understand the issue and the idea behind this PR.
My comment was only about moving `clkstarted_handle(dev, CLOCK_CONTROL_NRF_TYPE_LFCLK)` before `z_nrf_clock_calibration_lfclk_started()`. Is that actually needed?

Reply:
handling of lfclk started event changes state of the clock. If `z_nrf_clock_calibration_lfclk_started` is called earlier then it won't start calibration because clock is not yet started. If init was executed earlier then calibration will never be started.

Reply:
But `z_nrf_clock_calibration_lfclk_started` does not check the state of the clock. It only needs the `clock_dev` pointer to start the calibration. And this pointer is set in `z_nrf_clock_calibration_init` regardless of whether the clock is running or not. So I still don't see the need of this inversion of the order.